examples/model_compress/pruning/taylorfo_pruning_with_ddp_torch.py

```diff
@@ -0,0 +1,257 @@
+    dist.all_reduce(average_acc, op=dist.ReduceOp.SUM)
+    world_size = dist.get_world_size()
+    average_acc = average_acc / world_size
+
+    print('Average Accuracy: {}%\n'.format(average_acc.item()))
+    model.train(training)
+
+    return average_acc.item()
+
+def optimizer_scheduler_generator(model, _lr=0.1, _momentum=0.9, _weight_decay=5e-4, total_epoch=160):
+    optimizer = torch.optim.SGD(model.parameters(), lr=_lr, momentum=_momentum, weight_decay=_weight_decay)
+    scheduler = MultiStepLR(optimizer, milestones=[int(total_epoch * 0.5), int(total_epoch * 0.75)], gamma=0.1)
+    return optimizer, scheduler
+
+def retrain_model(
+    args,
+    model: nn.Module = None,
+):
+    # create an ddp model
+    if model is None:  # pretraining process
+        model = Mnist().cuda()
+        log_save_path = "pretraining.log"
+        model_save_path = "pretraining_best_model.pth"
+        epochs = args.pretrain_epochs
+        lr = args.pretraining_lr
+    else: # finetune process
+        log_save_path = "finetune.log"
+        model_save_path = "finetune_best_model.pth"
+        epochs = args.finetune_epochs
+        lr = args.finetune_lr
+
+    model = torch.nn.parallel.DistributedDataParallel(model, device_ids=[args.local_rank])
+
+    # create dataloaders
+    training_dataloader, validation_dataloader = create_dataloaders()
+    # create optimizer, lr_scheduler and criterion
+    optimizer, lr_scheduler = optimizer_scheduler_generator(model, \
+        _lr=lr, total_epoch=epochs)
+    criterion = torch.nn.CrossEntropyLoss()
+
+    # training and evaluation process
+    evaluation_func = functools.partial(evaluation, validation_dataloader)
+    best_acc = training(training_dataloader, model, optimizer, criterion, lr_scheduler,\
+        args.max_steps, epochs, args.local_rank, save_best_model=True, \
+        save_path=Path(args.log_dir) / model_save_path, \
+        log_path=Path(args.log_dir) / log_save_path, \
+        evaluation_func=evaluation_func)
+
+    # compute params and FLOPs
+    flops, params, _ = count_flops_params(model, torch.randn([32, 1, 28, 28]).cuda())
+
+    return flops, params, best_acc
+
+def pruned_model_process(args):
+    # load the pretrained model
+    model = Mnist().cuda()
+    state_dict = torch.load(Path(args.log_dir) / "pretraining_best_model.pth")
+    model.load_state_dict(state_dict)
+    model = torch.nn.parallel.DistributedDataParallel(model, device_ids=[args.local_rank])
+    # create dataloaders
+    training_dataloader, validation_dataloader = create_dataloaders()
+    # build a config_list
+    config_list = [{'total_sparsity': 0.7, 'op_types': ['Conv2d']}]
+    # create an evaluator
+    taylor_training = functools.partial(
+        training,
+        training_dataloader,
+        local_rank = args.local_rank,
+        log_path = Path(args.log_dir) / "taylor_pruning.log",
+        evaluation_func = functools.partial(evaluation, validation_dataloader),
+    )
```

Comment:
we don't need partial here, directly put these arguments into training func

Reply:
Ok, I will modify it.

Reply:
Ok, I will modify it.